### PR TITLE
 Use git ls-remote to track devel updates

### LINKED ISSUE #197: cannot unmarshal object into Go value of type main.branches on yay -Syu --devel
#### Affected Version

yay-git
#### Issue

On `yay -Syu --devel`
```
Cannot update 'cmus-git'
Error: json: cannot unmarshal object into Go value of type main.branches
Status code: 200
Body: {"id":7365136,"name":"cmus","full_name":"cmus/cmus","owner":{"login":"cmus","id":3125757,"avatar_url":"https://avatars3.githubusercontent.com/u/3125757?v=4","gravatar_id":"","url":"https://api.github.com/users/cmus","html_url":"https://github.com/cmus","followers_url":"https://api.github.com/users/cmus/followers","following_url":"https://api.github.com/users/cmus/following{/other_user}","gists_url":"https://api.github.com/users/cmus/gists{/gist_id}","starred_url":"https://api.github.com/users/cmus/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/cmus/subscriptions","organizations_url":"https://api.github.com/users/cmus/orgs","repos_url":"https://api.github.com/users/cmus/repos","events_url":"https://api.github.com/users/cmus/events{/privacy}","received_events_url":"https://api.github.com/users/cmus/received_events","type":"Organization","site_admin":false},"private":false,"html_url":"https://github.com/cmus/cmus","description":"Small, fast and powerful console music player for Unix-like operating systems.","fork":false,"url":"https://api.github.com/repos/cmus/cmus","forks_url":"https://api.github.com/repos/cmus/cmus/forks","keys_url":"https://api.github.com/repos/cmus/cmus/keys{/key_id}","collaborators_url":"https://api.github.com/repos/cmus/cmus/collaborators{/collaborator}","teams_url":"https://api.github.com/repos/cmus/cmus/teams","hooks_url":"https://api.github.com/repos/cmus/cmus/hooks","issue_events_url":"https://api.github.com/repos/cmus/cmus/issues/events{/number}","events_url":"https://api.github.com/repos/cmus/cmus/events","assignees_url":"https://api.github.com/repos/cmus/cmus/assignees{/user}","branches_url":"https://api.github.com/repos/cmus/cmus/branches{/branch}","tags_url":"https://api.github.com/repos/cmus/cmus/tags","blobs_url":"https://api.github.com/repos/cmus/cmus/git/blobs{/sha}","git_tags_url":"https://api.github.com/repos/cmus/cmus/git/tags{/sha}","git_refs_url":"https://api.github.com/repos/cmus/cmus/git/refs{/sha}","trees_url":"https://api.github.com/repos/cmus/cmus/git/trees{/sha}","statuses_url":"https://api.github.com/repos/cmus/cmus/statuses/{sha}","languages_url":"https://api.github.com/repos/cmus/cmus/languages","stargazers_url":"https://api.github.com/repos/cmus/cmus/stargazers","contributors_url":"https://api.github.com/repos/cmus/cmus/contributors","subscribers_url":"https://api.github.com/repos/cmus/cmus/subscribers","subscription_url":"https://api.github.com/repos/cmus/cmus/subscription","commits_url":"https://api.github.com/repos/cmus/cmus/commits{/sha}","git_commits_url":"https://api.github.com/repos/cmus/cmus/git/commits{/sha}","comments_url":"https://api.github.com/repos/cmus/cmus/comments{/number}","issue_comment_url":"https://api.github.com/repos/cmus/cmus/issues/comments{/number}","contents_url":"https://api.github.com/repos/cmus/cmus/contents/{+path}","compare_url":"https://api.github.com/repos/cmus/cmus/compare/{base}...{head}","merges_url":"https://api.github.com/repos/cmus/cmus/merges","archive_url":"https://api.github.com/repos/cmus/cmus/{archive_format}{/ref}","downloads_url":"https://api.github.com/repos/cmus/cmus/downloads","issues_url":"https://api.github.com/repos/cmus/cmus/issues{/number}","pulls_url":"https://api.github.com/repos/cmus/cmus/pulls{/number}","milestones_url":"https://api.github.com/repos/cmus/cmus/milestones{/number}","notifications_url":"https://api.github.com/repos/cmus/cmus/notifications{?since,all,participating}","labels_url":"https://api.github.com/repos/cmus/cmus/labels{/name}","releases_url":"https://api.github.com/repos/cmus/cmus/releases{/id}","deployments_url":"https://api.github.com/repos/cmus/cmus/deployments","created_at":"2012-12-29T10:05:50Z","updated_at":"2018-03-02T23:52:38Z","pushed_at":"2018-02-19T10:19:24Z","git_url":"git://github.com/cmus/cmus.git","ssh_url":"git@github.com:cmus/cmus.git","clone_url":"https://github.com/cmus/cmus.git","svn_url":"https://github.com/cmus/cmus","homepage":"https://cmus.github.io/","size":5790,"stargazers_count":2569,"watchers_count":2569,"language":"C","has_issues":true,"has_projects":true,"has_downloads":true,"has_wiki":true,"has_pages":false,"forks_count":291,"mirror_url":null,"archived":false,"open_issues_count":175,"license":{"key":"gpl-2.0","name":"GNU General Public License v2.0","spdx_id":"GPL-2.0","url":"https://api.github.com/licenses/gpl-2.0"},"forks":291,"open_issues":175,"watchers":2569,"default_branch":"master","organization":{"login":"cmus","id":3125757,"avatar_url":"https://avatars3.githubusercontent.com/u/3125757?v=4","gravatar_id":"","url":"https://api.github.com/users/cmus","html_url":"https://github.com/cmus","followers_url":"https://api.github.com/users/cmus/followers","following_url":"https://api.github.com/users/cmus/following{/other_user}","gists_url":"https://api.github.com/users/cmus/gists{/gist_id}","starred_url":"https://api.github.com/users/cmus/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/cmus/subscriptions","organizations_url":"https://api.github.com/users/cmus/orgs","repos_url":"https://api.github.com/users/cmus/repos","events_url":"https://api.github.com/users/cmus/events{/privacy}","received_events_url":"https://api.github.com/users/cmus/received_events","type":"Organization","site_admin":false},"network_count":291,"subscribers_count":122}
Cannot update 'kitty-git'
Error: json: cannot unmarshal object into Go value of type main.branches
Status code: 200
...
```

### LINKED ISSUE #205: yay_vcs can not handle packages with multiple VCS sources
I just started looking into moving over to the `git ls-remote` method and after looking through the code realised Yay can not handle packages with have multiple VCS sources in the packagebuild such as [brackets-git](https://aur.archlinux.org/packages/brackets-git/).

The easiest way to handle this is to change  `Info.URL` and `Info.SHA` to a `[]string` but this would make the old yay_vcs.json file incomparable with the new one. 

Maybe instead it would work if we allow duplicate package entries in the JSON. That would let us keep the format but would be a bit hanky and annoying to search through.

Instead I think we should move to a new file altogether `vcs.json` the `yay_` was a bit unneeded given that it was already in `~/.config/yay`. A new file would mean the user will start fresh but they can then `--gendb` their database back and it should work as before. The old file can be left unused without a problem or deleted by the user.

If we do decide on a new format I purpose something like this:
```json
{
	"brackets-git": [
		{
			"url": "https://api.github.com/repos/adobe/brackets-shell",
			"sha": "8ee733b4bf686e177a1e594616c22009c2bb777c"
		},
		{
			"url": "url2",
			"sha": "sha2"
		}
	],
	"another-package-git": [
		{
			"url": "url1",
			"sha": "sha1"
		}
	]
}
```

Also if we're moving files anyway maybe it would make more sense to have it in `~/.cache` than `~/.config`?

### PR DESCRIPTION
**Pull request by Morganamilo**
Use `git ls-remote <url> <branch>` to track devel updates.

This fixes #205, #79 and #181.
I assume this fixes #197 but I don't really know what caused that in the first place.

See commits and discussion in #205 for more info.

### CODE REVIEW

**Comment by Morganamilo** [comment]
@Jguer I'm Ready to merge this if you are. Also note this will add git as a dependency not just a make dependency.

File: vcs.go [hunk 1]
```diff
@@ -50,128 +33,127 @@ func createDevelDB() error {
 	return err
 }
 
-// parseSource returns owner and repo from source
-func parseSource(source string) (owner string, repo string) {
+// parseSource returns the git url and efault branch
```
Comment (Jguer) [inline vcs.go]: *default branch

just a typo

Comment (Morganamilo) [reply vcs.go]: I made that comment before i added the protocols support.

Fixed the typo and mentioned protocols.

**Comment by Jguer** [comment]
I mean, if people used `--devel`, I would assume they have git installed. Still I don't think it's a hard dependency and more of an optional dependency. `yay` will function without git using `--nodevel`

**Comment by Morganamilo** [comment]
Sure, add it as optional then. I would guess you would have to keep it in makedepends and also have it in optdepends then?

**Comment by Morganamilo** [comment]
Also now without the risk of rate limiting you can run `-Su --devel` as many times as you want with no problem so I was thinking devel could be enabled by default? If Git is not installed it should silently fail for the Git calls and work as normal apart from not making the vcs database.

**Comment by heyitswither** [comment]
@Morganamilo re: your comment on yay-git, git is a make depend, it's required to clone the package.

**Pull request merged**